src/decoder/mod.rs

```diff
@@ -53,23 +53,67 @@ impl OutputInfo {
     }
 }
 
+#[derive(Clone, Copy, Debug)]
+pub struct Limits {
+    /// max number of pixels: `width * height` (default: 67M = 2<sup>26</sup>)
+    pixels: u64,
+}
```

Comment:
I think there shouldn't be any problems with making `pixels` field public. This will also allow us to remove `set_pixels` setter.